modules/CameraCapture/CameraCapture.cpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file   AreaACameraAction.cpp
+ * @brief  カメラを制御するクラス
+ * @author Hara1274
+ */
+
+#include "CameraCapture.h"
+
+CameraCapture::CameraCapture()
+  : cap(),       // カメラデバイス（未接続で初期化）
+    cameraID(0)  // カメラIDを0で初期化
+{
+}
+
+int CameraCapture::findAvailableCameraID(int maxTested)
+{
+  for(int i = 0; i < maxTested; ++i) {
+    cv::VideoCapture cap(i);
+    if(cap.isOpened()) {
+      cap.release();
+      // 利用可能なIDを返す
+      if(cap.isOpened()) {
+        cap.release();
+        // 利用可能なIDを返す
+        std::cerr << "カメラID:" << i << " を使ってください" << std::endl;
+        return i;
+      }
+      std::cerr << "使用できるカメラIDが存在しません" << std::endl;
+      return i;
+    }
+  }
+  // どのIDも利用できなかった場合は-1を返す
+  return -1;
+}
+
+int CameraCapture::getCurrentCameraID()
+{
+  return cameraID;
+}
+
+bool CameraCapture::setCameraID(int id)
+{
+  if(id < 0) {
+    std::cerr << "無効なカメラIDです" << std::endl;
+    return false;
+  }
+  cameraID = id;
+  return true;
+}
+
+bool CameraCapture::openCamera()
+{
+  cap.open(cameraID);
+  if(!cap.isOpened()) {
+    std::cerr << "カメラを開くことができませんでした。" << std::endl;
+    return false;
+  }
+
+  return true;
+}
+
+void CameraCapture::setCapProps(double width, double height)
+{
+  cap.set(cv::CAP_PROP_FRAME_WIDTH, width);
+  cap.set(cv::CAP_PROP_FRAME_HEIGHT, height);
+}
+
+bool CameraCapture::getFrame(cv::Mat& outFrame)
+{
+  if(!cap.isOpened()) {
+    std::cerr << "カメラを開くことができませんでした。" << std::endl;
+    return false;
+  }
+  cap >> outFrame;
+  if(outFrame.empty()) {
+    std::cerr << "フレームの取得に失敗しました。" << std::endl;
+    return false;
+  }
+  return true;
+}
+
+bool CameraCapture::getFrames(cv::Mat* frames, int numFrames, int millisecondInterval)
+{
+  if(frames == nullptr || numFrames <= 0 || millisecondInterval <= 0) return false;
```

Comment:
```suggestion
  if (frames == nullptr) {
    std::cerr << "フレームバッファが null です。" << std::endl;
    return false;
  }

  if (numFrames <= 0) {
    std::cerr << "フレーム数が無効です: " << numFrames << std::endl;
    return false;
  }

  if (millisecondInterval <= 0) {
    std::cerr << "インターバルが無効です: " << millisecondInterval << "ms" << std::endl;
    return false;
  }
```
エラー処理を具体的に。こんな感じがいいかな？